tests/testthat/test-rand-forest-serde.R

```diff
@@ -0,0 +1,44 @@
+test_that("random forest classifier can be serialized and unserialized correctly", {
+  model <- cuda_ml_rand_forest(formula = Species ~ ., data = iris, trees = 200)
+  model_state <- cuda_ml_serialize(model)
+
+  data <- iris[-which(names(iris) == "Species")]
+
+  expected_preds <- predict(model, data)
+  expected_cls_probs <- predict(
+    model, data, output_class_probabilities = TRUE
+  )
+  actual_preds <- predict_in_sub_proc(
+    model_state,
+    data = data,
+    expected_mode = "classification",
+    expected_model_cls = "cuda_ml_rand_forest"
+  )
+  actual_cls_probs <- predict_in_sub_proc(
+    model_state,
+    data = data,
+    expected_mode = "classification",
+    expected_model_cls = "cuda_ml_rand_forest",
+    additional_predict_args = list(output_class_probabilities = TRUE)
+  )
+
+  expect_equal(expected_preds, actual_preds)
+  expect_equal(expected_cls_probs, actual_cls_probs, tol = 1e-4, scale = 1)
```

Comment:
At the moment RAPIDS cuML may have (I think) up to a single-precision ULP amount of rounding error introduced when reconstructing the rand forest model from the serialized state, which is why the new predictions are not exactly identical to the original ones, but are reasonably close.